src/gpu/intel/ocl/micro_sdpa.cpp

```diff
@@ -248,14 +286,28 @@ status_t micro_sdpa_t::pd_t::init_microkernels(impl::engine_t *engine) {
         case arch_t::xe_hpc:
         case arch_t::xe2:
         case arch_t::xe3:
-            config = choose_config_xehpc(
-                    d->head_size(), d->keys(), thin_q, quantized);
+            config = choose_config_xehpc(d->head_size(), d->keys(), thin_q,
+                    quantized, is_integrated);
         default: break;
     }
 
     if (!config) return status::unimplemented;
 
-    VDISPATCH_SDPA(config->unroll_n_kq * config->wg_n_kq
+    if (auto status = update_config_from_devenv_values(config, quantized)) {
```

Comment:
I wouldn't rely much on `status::success` equals to `0` or `false`.
And I would suggest not to assign `status` inside `if`. It's not generally a style the library uses.